packages/relay/src/lib/services/hapiService/hapiService.ts

```diff
@@ -0,0 +1,284 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import dotenv from 'dotenv';
+import findConfig from 'find-config';
+import { AccountId, Client, PrivateKey } from '@hashgraph/sdk';
+import { Logger } from 'pino';
+import { Registry, Counter } from 'prom-client';
+import { SDKClient } from '../../clients/sdkClient';
+import constants from '../../constants';
+import HbarLimit from '../../hbarlimiter';
+
+export default class HAPIService {
+  private transactionCount: number;
+  private errorCodes: number[];
+  private resetDuration: number;
+  private shouldReset: boolean;
+
+  private isReinitEnabled: boolean;
+  private isTimeResetDisabled: boolean;
+
+  private initialTransactionCount: number;
+  private initialErrorCodes: number[];
+  private initialResetDuration: number;
+
+  private hederaNetwork: string;
+  private clientMain: Client;
+
+  /**
+   * The SDK Client use for connecting to both the consensus nodes and mirror node. The account
+   * associated with this client will pay for all operations on the main network.
+   *
+   * @private
+   */
+  private client: SDKClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private logger: Logger;
+
+  /**
+   * This limiter tracks hbar expenses and limits.
+   * @private
+   */
+  private hbarLimiter: HbarLimit;
+  private readonly register: Registry;
+  private clientResetCounter: Counter;
+
+  /**
+   * @param {Logger} logger
+   * @param {Registry} register
+   */
+  constructor(logger: Logger, register: Registry, hbarLimiter: HbarLimit) {
+    dotenv.config({ path: findConfig('.env') || '' });
+
+    this.logger = logger;
+    this.hbarLimiter = hbarLimiter;
+
+    this.hederaNetwork = (process.env.HEDERA_NETWORK || '{}').toLowerCase();
+    this.clientMain = this.initClient(logger, this.hederaNetwork);
+    this.client = this.initSDKClient(logger, register);
+
+    const currentDateNow = Date.now();
+    this.initialTransactionCount = parseInt(process.env.HAPI_CLIENT_TRANSACTION_RESET!) || 0;
+    this.initialResetDuration = parseInt(process.env.HAPI_CLIENT_DURATION_RESET!) || 0;
+    this.initialErrorCodes = JSON.parse(process.env.HAPI_CLIENT_ERROR_RESET || "[]");
+
+    this.transactionCount = this.initialTransactionCount;
+    this.resetDuration = currentDateNow + this.initialResetDuration;
+    this.errorCodes = this.initialErrorCodes;
+
+    this.isReinitEnabled = true;
+    this.isTimeResetDisabled = this.resetDuration === currentDateNow;
+
+    if (this.transactionCount === 0 && this.errorCodes.length === 0 && this.isTimeResetDisabled) {
+      this.isReinitEnabled = false;
+    }
+    this.shouldReset = false;
+
+    this.register = register;
+    const metricCounterName = 'rpc_relay_client_service';
+    register.removeSingleMetric(metricCounterName);
+    this.clientResetCounter = new Counter({
```

Comment:
Q: should this metric also be a constructor param to work around the metric reset Fredy noted?

Reply:
No need, I've created this metric only to keep track of the SDK reinitializations. It's not used in the SDK client class.